debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1195,28 +1195,32 @@ private int checkJ9Object(J9ObjectPointer object, GCHeapRegionDescriptor regionD
 			}
 		}
 
-		if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
-
-			boolean isCorrectData = true;
-			J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
-			try {
-				if (J9IndexableObjectHelper.isInlineContiguousArraylet(array)) {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForContiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
-				} else {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForDiscontiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize())));
+		try {
+			if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
+				if (!_javaVM.isIndexableDataAddrPresent().isZero()) {
+					boolean isCorrectDataAddrPointer = true;
```

Comment:
The initialization is redundant; suggest the variable be declared immediately before the `if` statement where its value is defined.